Add Chat List

AnastasiaHolovash opened the pull request.

AnastasiaHolovash committed:
Add Chat List

AnastasiaHolovash committed:
Update ChatListCellView.swift to show unreadMessagesNumber

AnastasiaHolovash committed:
Merge remote-tracking branch 'origin/develop' into feature/chatList

# Conflicts:
#	Owl.xcodeproj/project.pbxproj
#	Owl/Modules/Login/EnterCode/EnterCodeStore.swift
#	Owl/Modules/Login/EnterPhone/EnterPhoneView.swift
#	Owl/Modules/Root/App/AppStore.swift

AnastasiaHolovash committed:
Update Colors

AnastasiaHolovash committed:
Merge branch 'develop' into feature/chatList

# Conflicts:
#	Owl.xcodeproj/project.pbxproj

AnastasiaHolovash committed:
Update Chats List fetching data subscription

ddanilyuk commented on Owl/Modules/Main/ChatList/ChatListCellView.swift:
```diff
@@ -0,0 +1,170 @@
+    }
+}
+
+// MARK: - Extensions
```
Maybe its better to call `// MARK: - Constants `?

ddanilyuk commented on Owl/Modules/Main/ChatList/ChatListView.swift:
```diff
@@ -0,0 +1,75 @@
+struct ChatListView: View {
+
+    var store: Store<ChatList.State, ChatList.Action>
+    @State private var searchText = ""
```
Remove this variable

ddanilyuk commented on Owl/Modules/Main/ChatList/ChatListView.swift:
```diff
@@ -0,0 +1,75 @@
+
+                    Button(
+                        action: { viewStore.send(.logout) },
+                        label: {
+                            Text("Logout")
+                        }
```
Maybe inline like action?

ddanilyuk commented on Owl/Modules/Main/MainStore.swift:
```diff
@@ -5,34 +5,128 @@
-        static let initialState = State()
+        var routes: IdentifiedArrayOf<Route<ScreenProvider.State>>
+
+        static let initialState = State(
```
Remove ChatList.State's from initialState.

AnastasiaHolovash committed:
Apply requested changes

- reorder files by name
- decompose ChatListCellView to ChatListCellView and ChatListCellStore
